libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java

```diff
@@ -868,23 +886,30 @@ private void handleFpsOnStop() {
     }
   }
 
+  private LocationObserver locationObserver = new LocationObserver() {
+    @Override
+    public void onRawLocationChanged(@NotNull Location rawLocation) {
+      Timber.d("raw location %s", rawLocation.toString());
+    }
+
+    @Override
+    public void onEnhancedLocationChanged(
+            @NotNull Location enhancedLocation,
+            @NotNull List<? extends Location> keyPoints
+    ) {
+      if (keyPoints.isEmpty()) {
+        updateLocation(enhancedLocation);
+      } else {
+        updateLocation((List<Location>) keyPoints);
+      }
+    }
+  };
+
   public void setCamera(Camera camera) {
     mapCamera.setCamera(camera);
   }
 
   public void setPuckDrawableSupplier(PuckDrawableSupplier supplier) {
     this.navigationPuckPresenter = new NavigationPuckPresenter(mapboxMap, supplier);
   }
-
-  public void updateCurrentLocationDrawable(@DrawableRes int gpsDrawable) {
```

Comment:
For documentation purposes noting that this breaks SemVer cc @langsmith @cafesilencio